### PR TITLE
Fix autoplay app crash bug

### PR DESCRIPTION
**Pull request by sddioulde**
### Description

Handle autoplay queue bug when only 1 track in a lineup. Also listen to mobile autoplay trigger. And some refactor.

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?

### How Has This Been Tested?

Updated relevant queue saga tests and also tested on browser and native mobile.

### How will this change be monitored?

For features that are critical or could fail silently please describe the monitoring/alerting being added.

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/sd-autoplay-fix

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/sd-autoplay-fix

File: src/store/queue/sagas.ts [hunk 1]
```diff
@@ -206,36 +258,22 @@ export function* watchNext() {
...
+      const uid = yield select(getUid)
+      const source = yield select(getSource)
+
+      yield call(handleQueueAutoplay, {
+        skip: !!skip,
+        ignoreSkip: false,
+        track
+      })
+
+      if (track) {
```
Comment (raymondjacobson) [inline src/store/queue/sagas.ts:276]: why the if track check here? is that effectively what we did with the index check before?

**Review by raymondjacobson — APPROVED** [review]
The change seems reasonable, but for testing, let's make sure all the following work

1. Autoplay works at end of lineup
2. Repeat single works at end of song that is first, middle, and end of lineup
3. Repeat full works at the end of line up
4. Pressing next manually when repeat single is on goes to the next track
5. Shuffle works within a lineup

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/sd-autoplay-fix

File: src/store/queue/sagas.ts (hunk 1 above)
Comment (sddioulde) [reply src/store/queue/sagas.ts:276]: > The change seems reasonable, but for testing, let's make sure all the following work
> 
> 1. Autoplay works at end of lineup
> 2. Repeat single works at end of song that is first, middle, and end of lineup
> 3. Repeat full works at the end of line up
> 4. Pressing next manually when repeat single is on goes to the next track
> 5. Shuffle works within a lineup

All the above work. Only weird thing I noticed is the playbar line doesn't reset on repeat single, which is an existing problem on master also. looking into it

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/sd-autoplay-fix

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/sd-autoplay-fix

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/sd-autoplay-fix

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/sd-autoplay-fix

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/sd-autoplay-fix

**Pull request merged**